webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/hooks/RunHooks.scala

```diff
@@ -300,8 +300,10 @@ object RunHooks {
    * Hooks must be executable and not ends with one of the
    * non-executable extensions.
    */
-   def getHooks(basePath: String, ignoreSuffixes: List[String]): Box[Hooks] = {
-     tryo {
+   def getHooks(basePath: String, ignoreSuffixes: List[String]): Box[Hooks] = getHooksPure(basePath, ignoreSuffixes).toBox
+
+   def getHooksPure(basePath: String, ignoreSuffixes: List[String]): IOResult[Hooks] = {
+     IOResult.effect {
```

Comment:
it seems this causes a huge drop in performance

Reply:
OK, I will need to test specifically that part. I think I spawn thread for non blocking part (the blocking part already spawn a thread via NuProcess. I need to check carefully